files/lib/chef_compat/copied_from_chef/chef/dsl/declare_resource.rb

```diff
@@ -94,6 +94,10 @@ def declare_resource(type, name, created_at=nil, run_context: self.run_context,
       def build_resource(type, name, created_at=nil, run_context: self.run_context, &resource_attrs_block)
         created_at ||= caller[0]
         Thread.exclusive do
+          begin
+            require 'chef/resource_builder' unless defined?(Chef::ResourceBuilder)
+          rescue LoadError
+          end
           require "chef_compat/copied_from_chef/chef/resource_builder" unless defined?(Chef::ResourceBuilder)
```

Comment:
problem is that if chef/resource_builder exists then we'll wind up with new code in declare_resource/etc hitting old code in the shipped chef/resource_builder.

not even quite certain how this fixes the problem, but it seems to try to fix the problem by defeating the purpose of the cookbook.